upload-api/stores/plans.js

```diff
@@ -1,12 +1,42 @@
 /**
  * 
  * @param {import("@web3-storage/w3infra-billing/lib/api").CustomerStore} customerStore
+ * @param {import('../types.js').BillingProvider} billingProvider
  * @returns {import("@web3-storage/upload-api").PlansStorage}
  */
-export function usePlansStore(customerStore) {
+export function usePlansStore(customerStore, billingProvider) {
   return {
-    get: async (customer) => {
-      const result = await customerStore.get({ customer })
+    initialize: async (account, externalID, plan) => {
```

Comment:
At first glance, this logic could be subject to a race condition. i.e. if two request come in in parallel and do `customerStore.get` and get a result indicating the customer does not exist, then both requests will move on to calling `customerStore.put` with similar values.

It would be helpful for a comment to acknowledge the seeming race condition and, if there are any mitigations, explain them. If there are no mitigations and it's just a known race condition, also good to have a comment for that.
(a similar race condition affected last stripe integration)

Reply:
ah true! added a comment: 

```js
      // Note that there is a possible race condition here - if two `initialize` calls come in at
      // the same time, they may both get here since the account hasn't been initialized yet.
      // In that case, the last `put` will win, which should be fine - as far as I can tell
      // there's no way we can really determine which one was "valid" so accepting whichever of them
      // happens to go last should be fine.
```